src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6538,13 +6538,24 @@ private void BindMemberAccessReportError(BoundMethodGroup node, BindingDiagnosti
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMemberOrExtensionNeedUsing, name, boundLeft.Type, plainName, "System");
                 }
+                else if (ShouldBeUsingAwaitForeach(boundLeft.Type, plainName))
+                {
+                    Error(diagnostics, ErrorCode.ERR_NoAwaitOnAsyncEnumerable, name, boundLeft.Type, plainName);
+                }
                 else
                 {
                     Error(diagnostics, ErrorCode.ERR_NoSuchMemberOrExtension, name, boundLeft.Type, plainName);
                 }
             }
         }
 
+        private bool ShouldBeUsingAwaitForeach(TypeSymbol receiver, string methodName)
+        {
+            // As per issue #53426: "Improve error message for awaiting an IAsyncEnumerable value".
+            return methodName == WellKnownMemberNames.GetAwaiter && receiver.IsIAsyncEnumerableType(Compilation);
```

Comment:
>methodName == WellKnownMemberNames.GetAwaiter [](http://example.com/codeflow?start=19&length=45)

There's a test for method name. Please include a test that covers the negative outcome of that test (ie. cover all branches of the added logic).